examples/config-retention.json

```diff
@@ -0,0 +1,68 @@
+{
+    "distSpecVersion": "1.1.0-dev",
+    "storage": {
+        "rootDirectory": "/tmp/zot",
+        "gc": true,
+        "gcDelay": "2h",
+        "gcInterval": "1h",
+        "retention": {
+            "dryRun": false,
+            "delay": "24h",
+            "policies": [
+                {
+                    "repositories": ["infra/*", "prod/*"],
```

Comment:
Can we add a check if same repository prefix patterns overlap in other rules? So that folks are informed about possibly "surprising" behaviors.

Would add this check/feedback for `zot verify` as well.

Just issue a warning that "found overlaps...etc" but move on.